src/NavItem.js

```diff
@@ -28,7 +28,7 @@ class NavItem extends React.Component {
   }
 
   handleClick(e) {
-    if (this.props.onSelect) {
+    if (this.props.onSelect && this.props.onSelect.name !== 'bound handleCollapse') {
```

Comment:
TBH i'm not sure why we even preventDefault here at all... @taion and idea? seems like bad practice to mess with any link